integration-tests/cucumber-tests-platform-smartmetering/src/test/java/org/opensmartgridplatform/cucumber/platform/smartmetering/glue/steps/database/device/DlmsDeviceSteps.java

```diff
@@ -616,4 +611,75 @@ private DeviceModel getDeviceModel(final Map<String, String> inputSettings) {
         return null;
     }
 
+    @When("^new keys are registered in the secret management database")
+    public void registerNewKeys(final Map<String, String> inputSettings) {
+        if (!inputSettings.containsKey(PlatformSmartmeteringKeys.DEVICE_IDENTIFICATION)) {
+            throw new IllegalArgumentException("No device identification provided");
+        }
+        final String deviceIdentification = inputSettings.get(PlatformSmartmeteringKeys.DEVICE_IDENTIFICATION);
+        final List<SecretType> secretTypesToCreate = Arrays
+                .asList(E_METER_AUTHENTICATION_KEY, E_METER_ENCRYPTION_KEY_UNICAST);
+        final List<String> keyTypeInputNames = secretTypesToCreate.stream().map(this::getKeyTypeInputName)
+                                                                  .collect(Collectors.toList());
+        if (Collections.disjoint(inputSettings.keySet(), keyTypeInputNames)) {
+            throw new IllegalArgumentException("None of the following keys provided: " + keyTypeInputNames.toString());
+        }
+        final DbEncryptionKeyReference encryptionKeyRef = this.encryptionKeyRepository
+                .findByTypeAndValid(EncryptionProviderType.JRE, new Date()).iterator().next();
+        for (int i = 0; i < secretTypesToCreate.size(); i++) {
+            if (inputSettings.containsKey(keyTypeInputNames.get(i))) {
+                final DbEncryptedSecret secret = new SecretBuilder()
+                        .withDlmsDevice(new DlmsDevice(deviceIdentification)).withSecretType(secretTypesToCreate.get(i))
+                        .withKey(inputSettings.get(keyTypeInputNames.get(i))).withSecretStatus(SecretStatus.NEW)
+                        .build();
+                secret.setEncryptionKeyReference(encryptionKeyRef);
+                this.encryptedSecretRepository.save(secret);
+            }
+        }
+    }
+
+    @Then("after 5 minutes, the new keys are recovered")
+    public void newKeysAreRecovered(final Map<String, String> inputSettings) throws InterruptedException {
+        final long waitTimeInMillis = 5 * 60 * 1000;
```

Comment:
The long wait appears to be necessary because of the key.recovery.delay (default in properties file 240000 millis), a delay which is used scheduling the key recovery process.
The only way to speed it up (I guess) would be if you could make sure a lower delay was used either for this specific test or in the test environment.
Given 5 minutes is a long time to hold the tests (even in the nightly builds) it might be worth checking if the delay can be set a lot lower in the test environment. Then, instead of sleeping for the duration of the wait time a quicker check can be done (possibly with retries after shorter intervals up to a maximum duration of 5 minutes, as the verification seems to just check the DB, and does not have serious side effects).